src/realm/object-store/sync/sync_session.cpp

```diff
@@ -938,10 +940,59 @@ void SyncSession::log_out()
             become_inactive(std::move(lock));
             break;
         case State::Inactive:
+        case State::Paused:
             break;
     }
 }
 
+void SyncSession::pause()
+{
+    util::CheckedUniqueLock lock(m_state_mutex);
+    switch (m_state) {
+        case State::Active:
+        case State::Dying:
+        case State::WaitingForAccessToken:
+            become_paused(std::move(lock));
+            break;
+        case State::Inactive:
```

Comment:
Is it correct to stay in Inactive here? This'd mean that pause() -> revive_if_needed() does sometimes result in an active session.

Reply:
No, it's not correct. become_paused() needs to check if the old state was inactive and early return in this case.